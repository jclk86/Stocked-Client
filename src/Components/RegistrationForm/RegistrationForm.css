.header_register {
  text-align: center;
  padding-top: 30px;
}

.register_logo {
  max-width: 100px;
}

.register_btn_container {
  text-align: center;
}

.label_registration {
  display: block;
  width: 85%;
  margin: 0 auto;
}

.message_redirect {
  padding-bottom: 50px;
}

.login_link {
  text-align: center;
}

.btn_login_link {
  text-decoration: none;
  color: rgba(46, 211, 189, 0.973);
  font-weight: bold;
}
