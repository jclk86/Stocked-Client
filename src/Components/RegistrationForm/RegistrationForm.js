import React, { Component } from "react";
import { Button, Input, Required, Form } from "../Utils/Utils";
import { NavLink } from "react-router-dom";
import "./RegistrationForm.css";

export default class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  };

  render() {
    return (
      <Form>
        <div className="header_register">
          <h2>Registration</h2>
        </div>
        <div className="full_name">
          <label
            htmlFor="RegistrationForm__full_name"
            className="label_registration"
          >
            Full name <Required />
          </label>
          <Input
            name="full_name"
            type="text"
            required
            id="RegistrationForm__full_name"
          />
        </div>
        <div className="user_name">
          <label
            htmlFor="RegistrationForm__user_name"
            className="label_registration"
          >
            User name <Required />
          </label>
          <Input
            name="user_name"
            type="text"
            required
            id="RegistrationForm__user_name"
          />
        </div>
        <div className="password">
          <label
            htmlFor="RegistrationForm__password"
            className="label_registration"
          >
            Password <Required />
          </label>
          <Input
            name="password"
            type="password"
            required
            id="RegistrationForm__password"
          />
        </div>
        <div className="email">
          <label
            htmlFor="RegistrationForm__email"
            className="label_registration"
          >
            Email
            <Required />
          </label>{" "}
          <Input
            name="email"
            type="text"
            required
            id="RegistrationForm__email"
          />
        </div>
        <div className="register_btn_container">
          <Button type="submit">Submit</Button>
        </div>
        <div className="login_link">
          <p>
            Already a user? <NavLink to={"/login"}>Login</NavLink>
          </p>
        </div>
      </Form>
    );
  }
}
