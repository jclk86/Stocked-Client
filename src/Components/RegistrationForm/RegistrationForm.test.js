import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RegistrationForm from "./RegistrationForm";
import sinon from "sinon";

describe(`Registration Form`, () => {
  it("Renders registration form without issues", () => {
    const wrapper = shallow(<RegistrationForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("fires submit", () => {
    const handleSubmitSpy = sinon.spy();
    const cb = () => {
      wrapper.find(".Form").simulate("submit");
    };
    const wrapper = mount(
      <BrowserRouter>
        <RegistrationForm onSubmit={handleSubmitSpy}></RegistrationForm>
      </BrowserRouter>
    );
    wrapper.setState({
      password: { value: "Password123!", touched: true },
      cb
    });
  });
});
