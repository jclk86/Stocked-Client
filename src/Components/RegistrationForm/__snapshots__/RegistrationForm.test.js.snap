// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form Renders registration form without issues 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
